bin.js

```diff
@@ -4,13 +4,22 @@ var proc = require('child_process')
 var os = require('os')
 var path = require('path')
 
+var targetMismatch = (process.env.npm_config_platform && process.platform !== process.env.npm_config_platform) ||
+  (process.env.npm_config_arch && process.arch !== process.env.npm_config_arch)
+
 if (!buildFromSource()) {
-  proc.exec('node-gyp-build-test', function (err, stdout, stderr) {
-    if (err) {
-      if (verbose()) console.error(stderr)
-      preinstall()
-    }
-  })
+  if (targetMismatch) {
+    // If the target is mismatched, then we can't test built output, and we can't build from source either.
+    // Instead, we just check whether a matching prebuild exists, and fail if it doesn't.
+    require('./index').path(process.cwd())
```

Comment:
```suggestion
    require('./index').path()
```

The dir argument is optional, so let's skip it, so that there's just one place where we define a default.